src/use-target-mutation/target-mutation-controller.ts

```diff
@@ -0,0 +1,20 @@
+import { Context, Controller } from 'stimulus'
+import { TargetMutationOptions, useTargetMutation } from './use-target-mutation'
+
+export class TargetMutationController extends Controller {
+
+  options!: TargetMutationOptions
+
+  constructor(context: Context) {
+    super(context)
+    requestAnimationFrame(() => {
+      const [observe, unobserve] = useTargetMutation(this, this.options)
+      Object.assign(this, { observe, unobserve })
+    })
+  }
+
+  observe!: () => {}
+  unobserve!: () => {}
+
+  [index: string]: any;
```

Comment:
Cool this is to somehow declare dynamic callbacks in TS?

Reply:
Kinda, that's what enabled this line:
```js
this.controller[targetCallback] && method(this.controller, targetCallback).call(this.controller, node)
```
Tells Typescript that the class accepts indexing its properties via string, and what type of thing is returned when you do that. 

https://www.typescriptlang.org/docs/handbook/advanced-types.html#index-types-and-index-signatures

If you decide to continue with https://github.com/stimulus-use/stimulus-use/pull/105 and just using `method(this.controller, targetCallback)`, then the need to have this declared goes away.